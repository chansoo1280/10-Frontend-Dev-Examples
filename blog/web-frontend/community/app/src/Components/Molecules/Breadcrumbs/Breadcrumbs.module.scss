@import "@Styles/scss/default.scss";

.breadcrumbs {

    &__breadcrumb {

        &,
        &:disabled {
            cursor: text;
            color: rgba($black, 0.45);
        }

        &--last,
        &--last:disabled,
        &--last:focus,
        &--last:hover {
            color: rgba($black, 0.85);
        }

        &--link:focus,
        &--link:hover {
            cursor: pointer;
            background-color: rgba($black, .06);
        }
    }
}