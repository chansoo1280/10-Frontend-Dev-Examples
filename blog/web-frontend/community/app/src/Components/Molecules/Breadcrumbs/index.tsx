// #region Global Imports
import React from "react"
import classNames from "classnames"
// #endregion Global Imports

// #region Local Imports
import { defaultProps, Space, Button } from "@Components"
import styles from "./Breadcrumbs.module.scss"
// #endregion Local Imports

export interface Breadcrumb {
    title: string
    href?: string
}
interface BreadcrumbsProps extends defaultProps {
    breadcrumbList: Breadcrumb[]
}

const Breadcrumbs = (props: BreadcrumbsProps): JSX.Element => {
    const { breadcrumbList, show, className, ...rest } = props
    const breadcrumbListLen = breadcrumbList.length
    const prefixCls = "breadcrumbs"
    const classes = classNames(
        styles[`${prefixCls}`],
        {
            [styles[`${prefixCls}--hide`]]: show === false,
        },
        className,
    )
    return (
        <Space separator="/" className={classes} {...rest}>
            {breadcrumbList.map((breadcrumb, idx) => (
                <Button
                    size="small"
                    type="text"
                    key={breadcrumb.title}
                    href={breadcrumb.href}
                    disabled={!breadcrumb.href}
                    className={classNames(styles[`${prefixCls}__breadcrumb`], {
                        [styles[`${prefixCls}__breadcrumb--last`]]: breadcrumbListLen - 1 === idx,
                        [styles[`${prefixCls}__breadcrumb--link`]]: breadcrumb.href,
                    })}
                >
                    {breadcrumb.title}
                </Button>
            ))}
        </Space>
    )
}
export default Breadcrumbs
