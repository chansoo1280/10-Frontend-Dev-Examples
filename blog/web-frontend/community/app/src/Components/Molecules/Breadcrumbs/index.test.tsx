import { fireEvent, render, screen } from "@testing-library/react"
import Breadcrumbs from "."
describe("Breadcrumbs", () => {
    it("renders breadcrumbs unchanged", () => {
        const { container } = render(<Breadcrumbs breadcrumbList={[{ title: "main" }, { title: "detail", href: "#a" }]} />)
        expect(container).toMatchSnapshot()
    })
    it("should render a breadcrumbs with the class", () => {
        render(<Breadcrumbs data-testid="breadcrumbs" breadcrumbList={[{ title: "main" }, { title: "detail", href: "#a" }]} />)

        const breadcrumbs = screen.getByTestId("breadcrumbs")
        expect(breadcrumbs).toBeInTheDocument()
        expect(breadcrumbs).toHaveClass("breadcrumbs")
    })
})
