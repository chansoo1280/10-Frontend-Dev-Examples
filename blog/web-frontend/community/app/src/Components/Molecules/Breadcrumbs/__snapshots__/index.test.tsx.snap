// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders breadcrumbs unchanged 1`] = `
<div>
  <div
    class="space breadcrumbs"
  >
    <button
      class="btn btn--small btn--text breadcrumbs__breadcrumb"
      data-click-animating="false"
      disabled=""
      type="button"
    >
      main
    </button>
    /
    <a
      class="btn btn--small btn--text breadcrumbs__breadcrumb breadcrumbs__breadcrumb--last breadcrumbs__breadcrumb--link"
      href="#a"
    >
      detail
    </a>
  </div>
</div>
`;
